test-apis/comprehensive_api/src/traits.rs

```diff
@@ -16,6 +16,21 @@ pub trait AssociatedType {
     type Type;
 }
 
+pub trait TraitReferencingOwnAssociatedType {
+    type OwnAssociatedType;
+
+    fn own_associated_type_output(&self) -> Self::OwnAssociatedType;
+    fn own_associated_type_output_explicit_as(
+        &self,
+    ) -> <Self as TraitReferencingOwnAssociatedType>::OwnAssociatedType;
+}
+
+pub trait TraitWithGenerics<T, U> {
+    type Foo;
+
+    fn bar() -> <Self as TraitWithGenerics<T, U>>::Foo;
+}
+
```

Comment:
There's currently no usage of a trait with an associated type in the api tests. Could we add that?

```suggestion
pub trait TraitReferencingOwnAssociatedType {
    type OwnAssociatedType;

    fn own_associated_type_output(&self) -> Self::OwnAssociatedType;
    fn own_associated_type_output_explicit_as(
        &self,
    ) -> <Self as TraitReferencingOwnAssociatedType>::OwnAssociatedType;
}

pub trait TraitWithGenerics<T, U> {
    type Foo;

    fn bar() -> <Self as TraitWithGenerics<T, U>>::Foo;
}

```

pub struct FooBar;

impl TraitReferencingOwnAssociatedType for FooBar {
	type OwnAssociatedType = ();
    ...
}

impl TraitWithGenerics<(), ()> for FooBar {
    type Foo = ();
	...
}
```

for example

Reply:
@Emilgardis Thanks for review! Would you be OK with you doing a follow-up PR that adds these tests? I think that is the best way to get the tests exactly as you want them :) I would be happy to review such a PR.

If that sounds OK, feel free to merge this PR on my behalf.

If you prefer me to add such tests before this PR is merged, that's also perfectly fine. Just let me know what you prefer.

Reply:
sounds good